File: Interfaces/IGrImportSerialLineService.cs
using WMS_WEBAPI.DTOs;

namespace WMS_WEBAPI.Interfaces
{
    public interface IGrImportSerialLineService
    {
        Task<ApiResponse<IEnumerable<GrImportSerialLineDto>>> GetAllAsync();
        Task<ApiResponse<GrImportSerialLineDto>> GetByIdAsync(long id);
        Task<ApiResponse<IEnumerable<GrImportSerialLineDto>>> GetByImportLineIdAsync(long importLineId);
        Task<ApiResponse<GrImportSerialLineDto>> CreateAsync(CreateGrImportSerialLineDto createDto);
        Task<ApiResponse<GrImportSerialLineDto>> UpdateAsync(long id, UpdateGrImportSerialLineDto updateDto);
        Task<ApiResponse<bool>> SoftDeleteAsync(long id);
        Task<ApiResponse<bool>> ExistsAsync(long id);
    }
}

File: WebApi/Interfaces/GoodReceiptInterfaces/IGrImportSerialLineService.ts
import type { CreateGrImportSerialLineDto, GrImportSerialLineDto, UpdateGrImportSerialLineDto } from '../../Models/index';
import type { ApiResponse, PagedResponse } from '../../Models/ApiResponse';

export interface IGrImportSerialLineService {
  getAll(): Promise<ApiResponse<GrImportSerialLineDto[]>>;
  getPaged(pageNumber: number, pageSize: number, sortBy: string, sortDirection: string): Promise<ApiResponse<PagedResponse<GrImportSerialLineDto>>>;
  getById(id: number): Promise<ApiResponse<GrImportSerialLineDto>>;
  getByImportLineId(importLineId: number): Promise<ApiResponse<GrImportSerialLineDto[]>>;
  create(createDto: CreateGrImportSerialLineDto): Promise<ApiResponse<GrImportSerialLineDto>>;
  update(id: number, updateDto: UpdateGrImportSerialLineDto): Promise<ApiResponse<GrImportSerialLineDto>>;
  softDelete(id: number): Promise<ApiResponse<boolean>>;
  exists(id: number): Promise<ApiResponse<boolean>>;
}
